baseapp/baseapp.go

```diff
@@ -60,6 +63,11 @@ type BaseApp struct { // nolint: maligned
 	idPeerFilter   sdk.PeerFilter   // filter peers by node ID
 	fauxMerkleMode bool             // if true, IAVL MountStores uses MountStoresDB for simulation speed.
 
+	// manages snapshots, i.e. dumps of app state at certain intervals
+	snapshotManager    *snapshots.Manager
+	snapshotInterval   uint64 // block interval between state sync snapshots
+	snapshotKeepRecent uint32 // recent state sync snapshots to keep
```

Comment:
In that case the manager would have to do the scheduling. Now, the baseapp figures out when to take snapshots (i.e. `if height%snapshotInterval==0`) and then calls `snapshotManger.Create()`. If the manager should do this, we would have to add a method `snapshotManager.MaybeCreate()` that would be called for every single committed height.

Considered implementing it that way originally, but thought it might be a bit odd - happy to change it.

Reply:
Ah, remembered why I didn't: then the snapshotManager would have to have a logger (to emit e.g. "Creating snapshot" messages). When we keep the policy outside, the manager doesn't have to do any logging, which is usually indicative of a cleaner design.

Reply:
I don't totally follow why having the manager have its own logger is a design flaw or poor design choice. But given this doesn't actually buy us anything, it's fine to leave as-is then.

Reply:
It couples the manager more tightly with the rest of the system. But I agree it's not really a big deal in this instance, either approach works for me.